### PR TITLE
Add Apidiff script

### PR DESCRIPTION
**Pull request by sedefsavas**
**What this PR does / why we need it**:
This PR adds apidiff script which is added to a job here: https://github.com/kubernetes/test-infra/pull/21451

Also adds PR-blocking tag to the CAPI's quick start test, it is added as PR-blocking presubmit job (https://github.com/kubernetes/test-infra/pull/21451)

**Checklist**:
- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests
- [ ] adds or updates e2e tests

Which issue(s) this PR fixes (optional, in fixes #<issue number>(, fixes #<issue_number>, ...) format, will close the issue(s) when PR gets merged):
Fixes https://github.com/kubernetes-sigs/cluster-api-provider-aws/issues/2130

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: scripts/ci-apidiff.sh [hunk 1]
```diff
@@ -0,0 +1,28 @@
+#!/bin/bash
+
+# Copyright 2020 The Kubernetes Authors.
```
Comment (richardcase) [inline scripts/ci-apidiff.sh:3]: Small nit, 2021

**Comment by randomvariable** [comment]
LGTM apart from the copyright date.

**Comment by sedefsavas** [comment]
Fixed the comment.

**Pull request merged**